sklearn/metrics/_classification.py

```diff
@@ -2556,6 +2556,7 @@ def balanced_accuracy_score(y_true, y_pred, *, sample_weight=None, adjusted=Fals
             "nan",
             StrOptions({"warn"}),
         ],
+        "separate_avg": ["boolean"],
```

Comment:
Let's rename to the following and keep it False by default to retain the current behavior. We will need to go through a deprecation cycle to change the default value to True.

```suggestion
        "structure_output": ["boolean"],
```